114 add a den view method

spencb-bit opened the pull request:
-adding DENview .cpp and .hpp files 
- needs to be properly linked to storage method, tested and reviewed.

spencb-bit committed:
Add files via upload

adding viewDEN.cpp manually bc of git issues

spencb-bit committed:
Add files via upload

adding .hpp directly bc of git issues

jaehyn99 commented on src/viewDEN.cpp:
```diff
@@ -0,0 +1,449 @@
+        updateWidth(maxValWidth, std::to_string(matrix.values[k]));
+
+    }
+
```
This seems like printing "I Index", "J Index", "Value" over numNonZeros with row Indices/ col Indices/values. It looks like coordinate list, rather than a dense view. It would be nice if I hear more explanation.

spencb-bit replied:
This is a fair point, will reformat this into dense view in next commit.

jaehyn99 reviewed (commented):
1. PR #155 configures and builds on Great Lakes (GL) with GCC and CMake as in the repo README.

2. ctest -N → no tests registered for this branch.

3. I ran a small COO smoketest. Construct SparseMatrix_COO<double>(3,3), call setValue 5× and assemble(). It runs successfully (prints “Hello from COO assemble”), final state reports 3×3 with nnz=5.

4. The branch contains src/viewDEN.cpp but it is not added to the library target, so the code does not compile by default. 

6. The implementation appears to assume a concrete DenseMatrix type, which is not present in this branch. As a result, the DEN view cannot be exercised yet.

spencb-bit committed:
Add files via upload

-modified viewDEN.cpp to be included in the SpMV namespace.
-changed the formatting of the output view to be a dense matrix.

spencb-bit committed:
Add files via upload

minor changes associated with .hpp file for viewDEN.cpp for namespace inclusion
